library/std/src/sys/pal/windows/mod.rs

```diff
@@ -114,6 +114,7 @@ pub fn decode_error_kind(errno: i32) -> ErrorKind {
         c::ERROR_TOO_MANY_LINKS => return TooManyLinks,
         c::ERROR_FILENAME_EXCED_RANGE => return InvalidFilename,
         c::ERROR_CANT_RESOLVE_FILENAME => return FilesystemLoop,
+        c::ERROR_LOCK_VIOLATION => return WouldBlock,
```

Comment:
On second thought, this is likely to be incorrect: this error can be returned from other operations such as opening a file that is write-locked. It would be better to revert to the old behavior of explicitly returning `ErrorKind::WouldBlock` only for `try_lock`.

Reply:
Ok, I reverted back to the old behavior. I see there's also some discussion in the [tracking issue](https://github.com/rust-lang/rust/issues/130994#issuecomment-2895342658) of also reverting back to the API with `TryLockError` and adding a `From` impl.